stingray/lightcurve.py

```diff
@@ -123,8 +140,9 @@ def __add__(self, other):
                                  "of same dimension and equal.")
 
```

Comment:
Okay, I have a question. When setting the `gti` keyword in instantiating a `Lightcurve` object, it automatically applies the GTIs, right? So the time stamps and counts left over are only those within the GTIs, also correct?

If so, then the example above should break this function, since here we assert that the time arrays of both light curves must be of same dimension and the time stamps exactly equal. Which means that the code below will never be executed. 

I think what you'd have to do is scrap the requirement that two light curves have to be equal to be added (and I think this might merit some discussion, because if they're not, and there are no GTIs, this could lead to some really wrong results). 

Then in order for this to work you'd (1) first have to compute the common GTIs, (2) then make two new light curves out of the old light curves with those new GTIs applied, (3) then add the counts of these together, and (4) *then* make a new light curve with the added counts.

If you do this, I think you might be able to check for consistency of time stamps after step (2), whereby you ensure that the remaining time stamps after applying GTIs are equal. I'd also suggest to throw a warning if the new GTIs aren't equal to the old GTIs, just so the user is warned that what will come out of this method is not a simple addition of the two light curves.